tachyon/math/polynomials/univariate/icicle/BUILD.bazel

```
@@ -0,0 +1,51 @@
+load("@local_config_cuda//cuda:build_defs.bzl", "if_cuda")
+load("//bazel:tachyon_cc.bzl", "tachyon_cc_library", "tachyon_cuda_library")
+
+package(default_visibility = ["//visibility:public"])
+
+tachyon_cc_library(
+    name = "fft_algorithm_conversion",
+    srcs = if_cuda(["fft_algorithm_conversion.cc"]),
+    hdrs = ["fft_algorithm_conversion.h"],
+    deps = [
+        "//tachyon:export",
+        "//tachyon/base:logging",
+        "//tachyon/math/polynomials/univariate:fft_algorithm",
+        "@icicle//:ntt",
+    ],
+)
+
+tachyon_cuda_library(
+    name = "icicle_ntt_bn254",
+    srcs = if_cuda(["icicle_ntt_bn254.cc"]),
+    hdrs = ["icicle_ntt_bn254.h"],
+    deps = ["@icicle//:ntt"] + if_cuda([
+        "@local_config_cuda//cuda:cudart_static",
+    ]),
+)
+
+tachyon_cc_library(
+    name = "icicle_ntt",
+    srcs = if_cuda(["icicle_ntt.cc"]),
+    hdrs = ["icicle_ntt.h"],
+    deps = [
+        ":icicle_ntt_bn254",
```

I think it would be better to add a direct dependency rather than a transitive dependency. What do you think?

```suggestion
        "@icicle//:ntt",
        ":icicle_ntt_bn254",
```